src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs

```diff
@@ -280,6 +278,17 @@ public bool HasNoResults
             get { return this.SearchResults.Count == 0; }
         }
 
+        /// <summary>
+        /// Checks if the user already has a package by the given name.
+        /// </summary>
+        /// <param name="packageName"></param>
+        /// <returns></returns>
+        internal bool UserAlreadyHasPackageInstalled(string packageName)
+        {
+            PackageManagerExtension packageManagerExtension = PackageManagerClientViewModel.PackageManagerExtension;
+            List<string> localPackageNames = packageManagerExtension.PackageLoader.LocalPackages.Select(x => x.Name).ToList();
+            return localPackageNames.Contains(packageName) || Downloads.Select(x => x.Name).Contains(packageName);
```

Comment:
I think you should also be checking package state - fyi @pinzart90 @sm6srw  thoughts?

Reply:
@mjkkirschner @pinzart @sm6srw When you say states, do you mean if the package is scheduled for uninstall etc?

Reply:
PackageLoadState.StateTypes hold the current load states
PackageLoadState.ScheduledTypes hold the scheduled types (ex. scheduled for deletion)
You should only have to do with PackageLoadState.StateTypes here...

Reply:
@pinzart90 Have modified the code to include a check against LoadState.State. 

https://github.com/DynamoDS/Dynamo/blob/0b9424dd795c84d47e793fc9338728d80184ae47/src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs#L286-L303